### DISCUSSION TITLE
arm64 assembler files instrumentation

### DISCUSSION THREAD

**Discussion by vfoltsgt** [AFLplusplus/AFLplusplus#1371] [Q&A]
Hello,

I have arm64 assembler (text) files I would like to instrument:
```
 AFL_AS_FORCE_INSTRUMENT=1 afl-gcc hello.s
```

This worked before for x64, but for aarch64 I'm getting:
```
 Error: unknown mnemonic `leaq' -- `leaq -(128+24)(%rsp),%rsp'
...
```

I dug into it and it seems like "afl-as.h" misses arm64 specifics so currently it tries to assemble the original aarch64 assembler mixed with x64 instrumentation code.

Are there any plans on supporting arm64?

**Comment by devnexen** [comment]
I do not think arm64 supports trampoline/ROP @vanhauser ?

**Comment by domenukk** [comment]
I don't know the use case, but maybe try frida_mode or qemu_mode instead? These don't need a compiler/smart Assembler and work on aarch64.
Alternatively look at how frida_mode does instrumentation and port it to a static pass

**Comment by vanhauser-thc** [comment]
there are no plans for arm (or other platform) support. it is an outdated way to instrument, but it is rather corner cases like yours where this would still be needed.
dominik's proposal for qemu_mode or frida_mode would be a possibility, or compiling that asm normally and not having instrumentation there (but in the rest of the project then) is another alternative. or you add arm64 support (and send a PR :) ).
